benchmarking/nursery/othpo/bo_warm_transfer.py

```diff
@@ -0,0 +1,225 @@
+    opt_val = task.objective_values(metric)[optima_idx[0], 0, 0]
+    return opt_val, set(
+        [
+            _make_config_str(keys_to_use, opt_idx, task.hyperparameters, de_str_dict)
+            for opt_idx in optima_idx
+        ]
+    )
+
+
+def get_optima_all_tasks(
+    transfer_learning_evaluations, opt_mode, keys_to_use, metric, nn, de_str_dict
+):
+    all_optima = []
+    for task_id in transfer_learning_evaluations:
+        opt_val, opt_configs = return_optima_from_task(
+            transfer_learning_evaluations[task_id],
+            opt_mode,
+            keys_to_use,
+            metric,
+            nn,
+            de_str_dict,
+        )
+        if opt_val is not None:
+            all_optima.append((task_id, opt_val, opt_configs))
+    return all_optima
+
+
+def get_sorted_optima(
+    transfer_learning_evaluations,
+    opt_mode,
+    keys_to_use,
+    metric,
+    nn,
+    de_str_dict,
+    sort_by_task_id,
+    shuffle_order,
+):
+
+    optimal_points = copy.deepcopy(
+        get_optima_all_tasks(
+            transfer_learning_evaluations,
+            opt_mode,
+            keys_to_use,
+            metric,
+            nn=nn,
+            de_str_dict=de_str_dict,
+        )
+    )
+
+    if sort_by_task_id:
+        # Sort so greater task_id comes first
+        optimal_points.sort(key=lambda tup: -tup[0])
+    else:
+        sign = -1 if opt_mode == "max" else 1
+        optimal_points.sort(key=lambda tup: sign * tup[1])
+
+    if shuffle_order:
+        np.random.shuffle(optimal_points)
+
+    return optimal_points
+
+
+def choose_warm_starters(
+    num_warm,
+    transfer_learning_evaluations,
+    opt_mode,
+    keys_to_use,
+    metric,
+    sort_by_task_id,
+    shuffle_order,
+):
+    points_to_evaluate = []
+    nn = 0
+
+    de_str_dict = {}
+
+    optimal_points = get_sorted_optima(
+        transfer_learning_evaluations,
+        opt_mode,
+        keys_to_use,
+        metric,
+        nn,
+        de_str_dict,
+        sort_by_task_id,
+        shuffle_order,
+    )
+
+    while len(optimal_points) > 0:
+        while len(points_to_evaluate) < num_warm and len(optimal_points) > 0:
+            task_to_add = optimal_points.pop(0)
+            config_to_add = task_to_add[2].pop()
+            if config_to_add not in points_to_evaluate:
+                points_to_evaluate.append(config_to_add)
+            if len(task_to_add[2]) > 0:  # still joint optima left
+                optimal_points.append(task_to_add)
+        nn += 1
+        if len(points_to_evaluate) == num_warm:
+            break
+
+        optimal_points = get_sorted_optima(
+            transfer_learning_evaluations,
+            opt_mode,
+            keys_to_use,
+            metric,
+            nn,
+            de_str_dict,
+            sort_by_task_id,
+            shuffle_order,
+        )
+
+    return [de_string_config(point, de_str_dict) for point in points_to_evaluate]
+
+
+class WarmStartBayesianOptimization(BoTorch):
```

Comment:
Nice one, not needed for this PR but we could put this one in `syne_tune/schedulers/` as I assume it could be generally useful.

Reply:
I'm tempted to add it here now and move it to syne_tune/schedulers with a more clear name once I have a bit more time.